src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -512,9 +512,45 @@ public async Task<HttpResponseMessage> SendAsyncCore(HttpRequestMessage request,
                 var response = new HttpResponseMessage() { RequestMessage = request, Content = new HttpConnectionResponseContent() };
                 ParseStatusLine(await ReadNextResponseHeaderLineAsync().ConfigureAwait(false), response);
 
-                // If we sent an Expect: 100-continue header, handle the response accordingly. Note that the developer
-                // may have added an Expect: 100-continue header even if there is no Content.
-                if (hasExpectContinueHeader)
```

Comment:
In removing this, it appears we can decrease the scope at which hasExpectContinueHeader is defined.  Can you move its declaration to L454?  That'll have a side effect of the compiler not needing to lift it to the state machine.

Reply:
> That'll have a side effect of the compiler not needing to lift it to the state machine.

Interesting, I don't know that local variable can be optimized by decreasing the scope. (Also, this field can be optimized without my change) Will fix.

Reply:
In async methods, if a "local" is written to prior to some await and then read after that await, it needs to be lifted to the state machine on the heap so that the value survives the async method getting suspended.  But if the value doesn't cross an await like that, it can just remain a local.